utils.R

```diff
@@ -0,0 +1,57 @@
+#' GetMissingColumns
+#' 
+#' Returns a character vector of missing required columns
+#' from a given dataframe
+#' 
+#' Required columns are:
+#' author, year, TP, FN, FP, TN
+#'
+#' @param df a dataframe
+#'
+#' @return a character vector of missing columns 
+#' (NULL if none are missing)
+GetMissingCols <- function(df) {
+    col_names <- names(df)
+    required_cols <- c("author", "year", "TP", "FN", "FP", "TN")
+    missing_cols <- c()
+    for (col in required_cols) {
+        if (!col %in% col_names) {
+            missing_cols <- c(missing_cols, col)
+        }
+    }
+    return(missing_cols)
+}
+
+#' IsValid
+#' 
+#' A function to validate that all the required columns
+#' are present in a dataframe
+#' see GetMissingColumns(df) for details
+#' on the required columns
+#'
+#' @param df a dataframe to be validated
+#'
+#' @return boolean - TRUE if there are no missing columns
+#' False if any columns are missing.
+IsValid <- function(df) {
```

Comment:
This whole function could be reduced to one line:
`return(is.null(GetMissingCols(df)))`

Reply:
This is deliberate, so that more statements can be added easily to add further validation, if needed. However, let me know if you want me to condense it for now.